From 68207d5623db9cbe52d2f201b77968af1879328d Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Sat, 22 Nov 2014 22:15:34 +0000
Subject: [PATCH 3/6] MinGW-w64: skip pthread time declaration\n\nI don't think
 this does what it says anymore.\nInstead it seems to be all about the Qt4
 integration,\n\n1. linking qt4 plugin to -lcomctl32 -luuid\n2. importing
 QWindowsIntegrationPlugin, AccessibleFactory\n    iff QT_STATICPLUGIN

---
 modules/gui/qt4/Makefile.am | 2 +-
 modules/gui/qt4/qt4.cpp     | 6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/modules/gui/qt4/Makefile.am b/modules/gui/qt4/Makefile.am
index c608999..1a564cf 100644
--- a/modules/gui/qt4/Makefile.am
+++ b/modules/gui/qt4/Makefile.am
@@ -20,7 +20,7 @@ libqt4_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(guidir)'
 if HAVE_OS2
 else
 if HAVE_WIN32
-libqt4_plugin_la_LIBADD += -lole32
+libqt4_plugin_la_LIBADD += -lole32 -lcomctl32 -luuid
 else
 libqt4_plugin_la_LIBADD += $(X_LIBS) $(X_PRE_LIB) -lX11
 endif
diff --git a/modules/gui/qt4/qt4.cpp b/modules/gui/qt4/qt4.cpp
index 1e25371..b6aef43 100644
--- a/modules/gui/qt4/qt4.cpp
+++ b/modules/gui/qt4/qt4.cpp
@@ -52,8 +52,10 @@
 #ifdef _WIN32 /* For static builds */
  #include <QtPlugin>
  #if HAS_QT5
-  Q_IMPORT_PLUGIN(QWindowsIntegrationPlugin)
-  Q_IMPORT_PLUGIN(AccessibleFactory)
+  #ifdef QT_STATICPLUGIN
+   Q_IMPORT_PLUGIN(QWindowsIntegrationPlugin)
+   Q_IMPORT_PLUGIN(AccessibleFactory)
+  #endif
  #else
   Q_IMPORT_PLUGIN(qjpeg)
   Q_IMPORT_PLUGIN(qtaccessiblewidgets)
-- 
2.1.3

